### PR TITLE
K8s yaml templates not rendered by k8sexecutor

### PR DESCRIPTION
**Pull request by dimberman**
There is a bug in the yaml template rendering caused by the logic that
yaml templates are only generated when the current executor is the
k8sexecutor. This is a problem as the templates are generated by the
task pod, which is itself running a LocalExecutor. Also generates a
"base" template if this taskInstance has not run yet.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by kaxil** [comment]
Tests need updating

File: airflow/models/renderedtifields.py [hunk 1]
```diff
@@ -102,7 +103,7 @@ def get_k8s_pod_yaml(cls, ti: TaskInstance, session: Session = None) -> Optional
             )
             .one_or_none()
         )
-        return result.k8s_pod_yaml if result else None
+        return result.k8s_pod_yaml if result else ti.render_k8s_pod_yaml()
```
Comment (ashb) [inline airflow/models/renderedtifields.py]: This shouldn't be done here -- it's breaking "encapsulation"/single responsibility.

This should be in the webserver view I think.

File: tests/kubernetes/test_pod_generator.py [hunk 2]
```diff
@@ -365,6 +365,7 @@ def test_reconcile_pods(self, mock_uuid):
                         image='',
                         name='name',
                         command=['/bin/command2.sh', 'arg2'],
+                        env=[],
```
Comment (ashb) [inline tests/kubernetes/test_pod_generator.py]: Is this change actually needed? It feels like it shouldn't be required, and the code needs to deal with not env not being specified if it doesn't.

```suggestion
```

File: tests/kubernetes/test_pod_generator.py [hunk 3]
```diff
@@ -473,6 +480,9 @@ def test_construct_pod_empty_executor_config(self, mock_uuid):
         worker_config.metadata.labels['app'] = 'myapp'
         worker_config.metadata.name = 'pod_id-' + self.static_uuid.hex
         worker_config.metadata.namespace = 'namespace'
+        worker_config.spec.containers[0].env.append(
+            k8s.V1EnvVar(name="AIRFLOW_IS_K8S_EXECUTOR_POD", value='True')
+        )
```
Comment (ashb) [inline tests/kubernetes/test_pod_generator.py:484]: This is static -- it should be added in to `pod_generator_base_with_secrets.yaml` instead of being added via code.
```suggestion
```

**Review by ashb — CHANGES_REQUESTED** [review]

File: tests/kubernetes/test_pod_generator.py (hunk 3 above)
Comment (dimberman) [reply tests/kubernetes/test_pod_generator.py:484]: @ashb This value is different as it is applied by the `construct_pod` method. I don't think it belongs in the pod_template_file as that would be confusing to anyone reading the code.

Comment (ashb) [reply tests/kubernetes/test_pod_generator.py:484]: Oh yes, you're right. I thought the file was the expected output, not the input. :+1:

**Review by kaxil — APPROVED** [review]

File: airflow/models/renderedtifields.py [hunk 4]
```diff
@@ -26,7 +27,7 @@
 from airflow.models.base import ID_LEN, Base
 from airflow.models.taskinstance import TaskInstance
 from airflow.serialization.helpers import serialize_template_field
-from airflow.settings import IS_K8S_OR_K8SCELERY_EXECUTOR, json
```
Comment (kaxil) [inline airflow/models/renderedtifields.py:29]: We should remove this variable from settings.py as it is no longer required

**Review by kaxil — CHANGES_REQUESTED** [review]

File: airflow/models/renderedtifields.py (hunk 4 above)
Comment (dimberman) [reply airflow/models/renderedtifields.py:29]: @kaxil I think it's used elsewhere

Comment (kaxil) [reply airflow/models/renderedtifields.py:29]: oh yea it still needs to be used in `airflow/www/extensions/init_jinja_globals.py` and `airflow/www/views.py` 

```
airflow/models/renderedtifields.py:from airflow.settings import IS_K8S_OR_K8SCELERY_EXECUTOR, json
airflow/models/renderedtifields.py:        if IS_K8S_OR_K8SCELERY_EXECUTOR:
airflow/settings.py:IS_K8S_OR_K8SCELERY_EXECUTOR = conf.get('core', 'EXECUTOR') in {
airflow/www/extensions/init_jinja_globals.py:from airflow.settings import IS_K8S_OR_K8SCELERY_EXECUTOR, STATE_COLORS
airflow/www/extensions/init_jinja_globals.py:            'k8s_or_k8scelery_executor': IS_K8S_OR_K8SCELERY_EXECUTOR,
airflow/www/views.py:        if not settings.IS_K8S_OR_K8SCELERY_EXECUTOR:
tests/models/test_renderedtifields.py:    @mock.patch("airflow.models.renderedtifields.IS_K8S_OR_K8SCELERY_EXECUTOR", new=True)
tests/models/test_taskinstance.py:    @patch("airflow.models.renderedtifields.IS_K8S_OR_K8SCELERY_EXECUTOR", new=True)
tests/www/test_views.py:        with mock.patch.object(settings, "IS_K8S_OR_K8SCELERY_EXECUTOR", True):
```

**Review by kaxil — APPROVED** [review]

**Review by ashb — APPROVED** [review]
